terraform/variables.tf

```diff
@@ -0,0 +1,159 @@
+variable "model" {
+  type    = string
+  default = "jimm"
+}
+
+variable "include_oauth-external-idp-integrator" {
+  description = <<EOT
+  If you set this to true, you need to configure the oauth-external-idp-integrator charm.
+  In this way you can use the JIMM charm without the iam-bundle.
+  EOT
+  type        = bool
+  default     = false
+}
+
+variable "name" {
+  description = "JIMM application name"
+  type        = string
+  default     = "jimm"
+}
+
+variable "trust" {
+  description = "The status to grant the Juju application full access to the cluster."
+  type        = bool
+  default     = true
+}
+
+
+variable "charm" {
+  description = "The application charm operator information."
+  type = object({
+    name : string
+    channel : string
+    base : string
+  })
+  default = {
+    name    = "juju-jimm-k8s"
+    channel = "3/edge"
+    base    = "ubuntu@22.04"
+  }
+}
+
+// The JIMM charm configuration
+// More info at https://charmhub.io/juju-jimm-k8s/configurations
+variable "jimm_config" {
+  type = object({
+    uuid              = optional(string, "")
+    controller_admins = optional(string, "")
+    log_level         = optional(string, "info")
+    dns_name          = string
+    public_key        = optional(string, "eaj1OMeHsd8OyDTqE+OOpkmWCNd1Py6mF1i3UD+VoCk=") // DO NOT USE IN PRODUCTION
+    private_key       = optional(string, "vx+aEalWSM524h9NAZcyZlUFvOLkiRN2J7K5uz9FW1c=") // DO NOT USE IN PRODUCTION
+  })
+  description = <<EOT
+    jimm_config = {
+      uuid: "The UUID advertised by the JIMM controller. If not provided, one will be generated for you."
+      controller_admins: "Whitespace separated list of candid users (or groups) that are made controller admins by default."
+      log_level: "Level to out log messages at, one of debug, info, warn, error, dpanic, panic, and fatal."
+      dns_name: "DNS hostname that JIMM is being served from."
+      public_key: "The public part of JIMM's macaroon bakery keypair."
+      private_key: "The private part of JIMM's macaroon bakery keypair."
+    }
+  EOT
+}
+
+variable "units" {
+  description = "Number of JIMM units. Default to 3 for HA."
+  type        = number
+  default     = 3 #
+}
+
+variable "oauth-external-idp-integrator_application_name" {
```

Comment:
perhaps create a structure variable so it is consistent with the one for JIMM charm